Event.js

```diff
@@ -0,0 +1,138 @@
+import * as L from "./Logger.js";
+import * as Validate from "./Validators.js";
+
+class Event {
+    #state;
+
+    constructor(state) {
+        this.#state = state;
+    }
+
+    /**
+     * Used to record an event;
+     *
+     * @param {string} eventName event name.
+     * @param {Segmentation} segments event segmentation.
+     * @param {number} eventCount event count.
+     * @param {number} eventSum event sum.
+     * @return {void} void
+     */
+    recordEvent(eventName, segments, eventCount, eventSum) {
+        if (!this.#state.isInitialized) {
+            L.e("recordEvent, 'init' must be called before 'recordEvent'");
+            return;
+        }
+        if (!eventName) {
+            L.e("recordEvent, eventName is required");
```

Comment:
all of  logs here to debug, and for striungvlalidate too, and inside these methods u dont have to pring logs